codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/specs/JsonProtocolSpec.java

```diff
@@ -94,7 +94,8 @@ public MethodSpec initProtocolFactory(IntermediateModel model) {
                       .addCode(".defaultServiceExceptionSupplier($T::builder)\n", baseException)
                       .addCode(".protocol($T.$L)\n", AwsJsonProtocol.class, protocolEnumName(metadata.getProtocol()))
                       .addCode(".protocolVersion($S)\n", metadata.getJsonVersion())
-                      .addCode("$L", customErrorCodeFieldName());
+                      .addCode("$L", customErrorCodeFieldName())
+                      .addCode("$L", hasAwsQueryCompatible());
```

Comment:
can we conditionally add this only if hasAwsQueryCompatible is true similar to V1 ?

Reply:
Please run unit test case codegen module , this change would have cause failures in existing test cases